src/subscription-user/dto/update-subscription-user.dto.ts

```diff
@@ -0,0 +1,16 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsBoolean, IsDefined, IsNotEmpty, IsString } from 'class-validator';
+
+export class UpdateSubscriptionUserDto {
+  @IsBoolean()
+  @IsNotEmpty()
+  @IsDefined()
+  @ApiProperty({ type: Boolean })
+  isActive: boolean;
```

Comment:
Optional - is this 'active' in the database ? If so - might be better to keep it consistent

Reply:
Right now there's a `cancelledAt` date - I thought it would be good to keep the date for audit reasons. 

Do you mean instead use a boolean `isActive`?

Reply:
e.g. how long does a user keep a whatsapp subscription could be an interesting data point 

If we're having a cancelledAt date, then having an isActive column seemed redundant to me

Reply:
oh i see. I see - I think I had it in my head this was a pure "update subscription" end point. So whatever is in the body is whats being updated - does that make sense? i.e. 
```
method: PUT/PATCH
endpoint: /subscription/:id
body: {
cancelledAt: DATE,
}
```

As there seems to be a desire to have more granular control over what we can update. A specific endpoint could make sense. Perhaps adding just the cancelledAt: Date as an option might make sense. Then there is no need to be checking if it is active or not. 

```
method: PUT/PATCH
url: /subscriptions/whatsapp/:id
body: {cancelledAt: Date}
```
I guess I am being a bit purist about REST here. Trying to keep the content of the body as close as we can to what is being updated. 

This is all optional as I can see the argument for the way you have done it.

Reply:
@eleanorreem agree with what youre saying - having the update DTO reflect the entity that it's updating makes a lot more sense. 

Have changed the code so that the id is in the URL and the param takes in a cancelledAt date. Let me know your thoughts.

Reply:
looks great ⭐